File: dataset_reduzido/avatar/Python/Code/codeforces_629_B.py
import sys
input = sys.stdin.readline
N = 368
m = [ 0 ] * N
f = [ 0 ] * N
for i in range ( int ( input ( ) ) ) :
    x , a , b = input ( ) [ : - 1 ].split ( )
    a = int ( a )
    b = int ( b ) + 1
    if x == 'M' :
        m [ a ] += 2
        m [ b ] -= 2
    else :
        f [ a ] += 2
        f [ b ] -= 2
a , b , c = 0 , 0 , 0
for i in range ( N ) :
    a += m [ i ]
    b += f [ i ]
    if min ( a , b ) > c :
        c = min ( a , b )
print ( c )

File: output/LLaMa/avatar/Python/Java/codeforces_629_B.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = 368;
        int[] m = new int[N];
        int[] f = new int[N];
        int i = scanner.nextInt();
        while (scanner.hasNextLine()) {
            String[] input = scanner.nextLine().split(" ");
            int x = input[0].equals("M") ? 1 : 0;
            int a = Integer.parseInt(input[1]) - 1;
            int b = Integer.parseInt(input[2]);
            m[a] += x * 2;
            m[b] -= x * 2;
            f[a] += (1 - x) * 2;
            f[b] -= (1 - x) * 2;
        }
        int a = 0, b = 0, c = 0;
        for (int j = 0; j < N; j++) {
            a += m[j];
            b += f[j];
            c = Math.min(a, b) > c ? Math.min(a, b) : c;
        }
        System.out.println(c);
    }
}